packages/router/src/router.tsx

```diff
@@ -251,9 +258,12 @@ const NotFoundChecker: React.FC<{ children: React.ReactNode }> = ({
         )
 
         if (match) {
+          // name will be undefined for redirect routes
+          routeName = name || ''
```

Comment:
Is there a case where we wouldn't have a routeName, and therefore `|| ''`?

Reply:
Yes, isn't the reason clear from the comment?

Reply:
Ah, I was reading that as `undefined`, not an empty value!

Reply:
In other words... I was thinking `routeName` will be `undefined` for redirect routes, but then was left wondering why we made it an empty string.

Reply:
You're right. I'll change the code a bit, try to make it easier to understand

Reply:
Discussion is continued in the other comment on the same line